src/3rd_party_update.c

```diff
@@ -170,66 +159,23 @@ enum swupd_code thir_party_update_main(int argc, char **argv)
 		print_help();
 		return SWUPD_INVALID_OPTION;
 	}
-	progress_init_steps("3rd-party-update", steps_in_update);
 
 	ret_code = swupd_init(SWUPD_ALL);
 	if (ret_code != SWUPD_OK) {
 		error("Failed swupd initialization, exiting now\n");
-		goto exit;
+		return ret_code;
 	}
-
-	/* Update should always ignore optional bundles */
-	globals.skip_optional_bundles = true;
+	progress_init_steps("3rd-party-update", steps_in_update);
 
 	/* set the command options */
 	update_set_option_version(cmdline_option_version);
 	update_set_option_download_only(cmdline_option_download_only);
 	update_set_option_keepcache(cmdline_option_keepcache);
 
-	/* load the existing 3rd-party repos from the repo.ini config file */
-	repos = third_party_get_repos();
-
-	/* backup the original state_dir and path_prefix values */
-	state_dir = strdup_or_die(globals.state_dir);
-	path_prefix = strdup_or_die(globals.path_prefix);
+	/* update 3rd-party bundles */
+	ret_code = third_party_run_operation_multirepo(cmdline_option_repo, update_repos, SWUPD_NO);
```

Comment:
expected_ret_Code = SWUPD_NO??

Reply:
if you run "update" on a system that is up to date you will get SWUPD_NO as return code. If an update is needed and performed correctly then SWUPD_OK is returned. In this case if any repo does an update it will return SWUPD_OK. Both of the codes are expected by the way.

Reply:
Make sense. Thanks for explaining.

Reply:
you know what? I think I may be mistaken, I just tried running update in my system and it returned SWUPD_OK in both cases, when the system updates and when the system is up to date.... I'll double check, and if I'm wrong I'll update this PR with SWUPD_OK as expected_ret_code

Reply:
confirmed, update returns SWUPD_OK when it updates and when it is already up to date. I'll update the PR. Thanks for noticing it @karthikprabhuvinod

Reply:
when I was updating this in the code I realized this was setup like this so the "update --status" command works properly, that command expects SWUPD_NO if there are no updates available. When the --status flag is not being used we are overwriting the return code so it will work properly too. So it is fine as it is. 😆